### ISSUE TITLE
Garry´s Mod Server update function is broken (./gmodserver update)

### ISSUE THREAD

**Issue by dev-lukas** [GameServerManagers/LinuxGSM#3106]
## User Story
gmodserver update function doesn´t work at the moment. update process everytime looks like this:

[ OK ] Updating gmodserver: Checking local build: SteamCMD
[ FAIL ] Updating gmodserver: Checking remote build: SteamCMD

I triple checked my firewall and dependencies, every needed port is open.

## Information about my system:
Distro Details
=================================
Distro:    Ubuntu 18.04.5 LTS
Arch:      x86_64
Kernel:    4.15.0-88-generic
Hostname:  Ubuntu-1804
Uptime:    192d, 14h, 0m
tmux:      tmux 2.6
glibc:     2.27

Server Resource
=================================
CPU         
Model:      AMD Ryzen 5 3600 6-Core Processor
Cores:      12
Frequency:  2120.548MHz
Avg Load:   0.31, 0.30, 0.29

Memory     
Mem:       total  used   free  cached  available
Physical:  63GB   7.9GB  55GB  32GB    55GB
Swap:      32GB   0B     32GB

Storage
Filesystem:  /dev/md2
Total:       437G
Used:        30G
Available:   385G

Network
Interface:   enp34s0
Link Speed:  1000Mb/s
IP:           [censored]

Game Server Resource Usage
=================================
CPU Used:  0.8%
Mem Used:  0.2%  179MB

Storage
Total:        7.5G
Serverfiles:  6.6G
Backups:      100K

Garry's Mod Server Details
=================================
Server name:  [censored]      
App ID:           4020
Server IP:        [censored]:27015
Server password:  ********
RCON password:    ********
Players:          0/16
Current map:      ttt_bb_teenroom_b2
Default map:      ttt_bb_teenroom_b2
Game mode:        terrortown
Tick rate:        66
Master server:    listed
Status:           ONLINE

gmodserver Script Details
=================================
Script name:            gmodserver
LinuxGSM version:       v20.5.1
glibc required:         2.15
Discord alert:          off
Email alert:            off
IFTTT alert:            off
Mailgun (email) alert:  off
Pushbullet alert:       off
Pushover alert:         off
Rocketchat alert:       off
Slack alert:            off
Telegram alert:         off
Update on start:        off
User:                   gmodserver
Location:               /home/gmodserver
Config file:            /home/gmodserver/serverfiles/garrysmod/cfg/gmodserver.cfg

Backups
=================================
No Backups created

Command-line Parameters
=================================
./srcds_run -game garrysmod -strictportbind -ip  [censored] -port 27015 -tickrate 66 +host_workshop_collection 2034936295 -authkey ******** +clientport 27005 +tv_port 27020 +gamemode terrortown +map ttt_bb_teenroom_b2 +sv_setsteamaccount ******** +servercfgfile gmodserver.cfg -maxplayers 16 -disableluarefresh

DESCRIPTION  DIRECTION  PORT   PROTOCOL
> Game/RCON  INBOUND    27015  tcp/udp
> SourceTV   INBOUND    27020  udp
< Client     OUTBOUND   27005  udp
Status:	ONLINE

All relevant logs that I could find:

[dev-debug-function-order.log](https://github.com/GameServerManagers/LinuxGSM/files/5573144/dev-debug-function-order.log)
[dev-debug.log](https://github.com/GameServerManagers/LinuxGSM/files/5573145/dev-debug.log)
[steamlogs.zip](https://github.com/GameServerManagers/LinuxGSM/files/5573155/steamlogs.zip)

**Comment by h3o66** [comment]
Thanks for attaching that much logs.
But the dev-debug.log does not seem to include the needed debugged command update in this case, here was only the `dev` command executed.

What i can see from the steam logs: 

Content log:
```
[2020-10-27 07:00:15] HTTP (SteamCache,66) - cache8-sto2.steamcontent.com ([2a01:bc80:8:100::9b85:fc09]:443 / [2a01:bc80:8:100::9b85:fc09]:443, host: cache8-sto2.steamcontent.com): CDepotHTTPDownloadInterface::BYldTestConnection - timed out
```

there seems to be a timeout with the ipv6 connection at least, in the other logs it did not seem to be a problem.

maybe try as well to verify the gamefiles (save all changes before doing this):
`./gmodserver validate`

**Comment by dev-lukas** [comment]
Hi,
I redid the dev debug logs now, deleting old ones and doing now 3 update calls inbetween enabling and disabling dev mode.
Console looked as following while doing that:

[dev-debug.log](https://github.com/GameServerManagers/LinuxGSM/files/5573999/dev-debug.log)
[dev-debug-function-order.log](https://github.com/GameServerManagers/LinuxGSM/files/5574000/dev-debug-function-order.log)
![dev](https://user-images.githubusercontent.com/45562796/99809249-4a574000-2b42-11eb-90f7-4c8a9af8ce02.PNG)

 Validating didnt do much at at all, doing that looked like that and was succesfull in under 30 seconds, updating is still not working:

![validate](https://user-images.githubusercontent.com/45562796/99809286-5c38e300-2b42-11eb-8129-993a07719f5e.PNG)

Its probably just faster to do a clean reinstall of the gameserver huh?

**Comment by h3o66** [comment]
The dev-debug.log is not held persistent, so in this case you need to do it this way:
```
Enable dev debug
execute function.
backup logs for upload or upload them directly.
disable dev debug
```

You can as well contact me on discord (nick: extreme1st#7327 ) to debug this a bit better, maybe join the discord of lgsm for this.

**Comment by JoinedSenses** [comment]
Possibly related to #3129.

**Comment by h3o66** [comment]
Should be fixed with the latest hotfix v20.6.2
